virtual_ecosystem/models/soil/soil_model.py

```diff
@@ -208,15 +220,19 @@ def integrate(self) -> dict[str, DataArray]:
             [
                 self.data[name].to_numpy()
                 for name in map(str, self.data.data.keys())
-                if name.startswith("soil_c_pool_") or name.startswith("soil_enzyme_")
+                if name.startswith("soil_c_pool_")
+                or name.startswith("soil_enzyme_")
+                or str(name).startswith("soil_n_pool_")
```

Comment:
This should not be needed, if it is not needed for the others. 
```suggestion
            if name.startswith("soil_c_pool_")
            or name.startswith("soil_enzyme_")
            or name.startswith("soil_n_pool_")
```

Reply:
I think there's an argument for having an explicit list of these variables somewhere. Partly that's to avoid having repeated string matches, but I also think it's clearer if the model declares the set of of things it cares about.

I think you could declare it as an additional class attribute (don't think the ABC forbids that?) or it could be a GLOBAL variable within the module. Sure, you'd have to maintain it, but that seems better?

Reply:
Yeah this is a sensible suggestion, particularly as this is basically the list of variables that get altered by the integrator (everything else that gets used from the data object is fixed for any given integration run).

Reply:
This list is actually identical to `vars_updated` (because everything in the soil model is updated by the integrator) so it probably makes sense to just use that (if I can figure out how to pass it everywhere it needs to go)